addons/analytic/models/analytic_line.py

```diff
@@ -73,20 +80,57 @@ class AccountAnalyticLine(models.Model):
         store=True,
         compute_sudo=True,
     )
-    plan_id = fields.Many2one(
-        'account.analytic.plan',
-        related='account_id.plan_id',
-        store=True,
-        readonly=True,
-        compute_sudo=True,
-    )
     category = fields.Selection(
         [('other', 'Other')],
         default='other',
     )
 
-    @api.constrains('company_id', 'account_id')
-    def _check_company_id(self):
+    def _compute_auto_account(self):
+        plan = self.env['account.analytic.plan'].browse(self.env.context.get('analytic_plan_id'))
+        for line in self:
+            line.auto_account_id = bool(plan) and line[plan._column_name()]
+
+    def _inverse_auto_account(self):
         for line in self:
-            if line.account_id.company_id and line.company_id.id != line.account_id.company_id.id:
-                raise ValidationError(_('The selected account belongs to another company than the one you\'re trying to create an analytic item for'))
+            line[line.auto_account_id.plan_id._column_name()] = line.auto_account_id
+
+    def _search_auto_account(self, operator, value):
+        project_plan, other_plans = self.env['account.analytic.plan']._get_all_plans()
+        return OR([
+            [(plan._column_name(), operator, value)]
+            for plan in project_plan + other_plans
+        ])
+
+    def _get_view(self, view_id=None, view_type='form', **options):
+        arch, view = super()._get_view(view_id, view_type, **options)
+        if self.env['account.analytic.plan'].check_access_rights('read', raise_exception=False):
+            project_plan, other_plans = self.env['account.analytic.plan']._get_all_plans()
+
+            # Find main account nodes
+            account_node = next(iter(arch.xpath('//field[@name="account_id"]')), None)
+            account_filter_node = next(iter(arch.xpath('//filter[@name="account_id"]')), None)
+
+            # Force domain on main account node as the fields_get doesn't do the trick
```

Comment:
I have no idea why. Adding the domain in `fields_get` is not reflected for `account_id` in search views without this :shrug: 
I guess it is because of some caching somewhere